java-squid/src/main/java/org/sonar/java/se/ConstraintManager.java

```diff
@@ -107,7 +143,119 @@ BooleanConstraint inverse() {
     }
   }
 
+  public ProgramState processNewInstance(ProgramState initialState, NewClassTree syntaxNode) {
+    ProgramState.Pop pop = initialState.unstackValue(syntaxNode.arguments().size());
+    ProgramState previousState = pop.state;
+    List<SymbolicValue> arguments = pop.values;
+    final SymbolicValue value;
+    final NullConstraint constraint;
+    if (isOpeningResource(syntaxNode)) {
+      final CloseableArgument closeableArgument = getCloseableArgument(syntaxNode, arguments);
+      if (closeableArgument == null) {
+        value = new SymbolicValue(counter++, syntaxNode);
+        constraint = NullConstraint.OPENED;
+      } else if (closeableArgument.value == null) {
+        value = createSymbolicValue(syntaxNode);
+        constraint = NullConstraint.NOT_NULL;
+      } else {
+        final SymbolicValue argumentValue = closeableArgument.value;
+        if (previousState.getConstraint(argumentValue) != NullConstraint.OPENED) {
+          previousState = previousState.addConstraint(argumentValue, NullConstraint.OPENED);
+        }
+        value = new SymbolicValue.ResourceWrapperSymbolicValue(counter++, argumentValue);
+        constraint = NullConstraint.NOT_NULL;
+      }
+    } else {
+      value = createSymbolicValue(syntaxNode);
+      constraint = NullConstraint.NOT_NULL;
+    }
+    ProgramState programState = previousState.stackValue(value);
+    return value.setSingleConstraint(programState, constraint);
+  }
+
+  private static CloseableArgument getCloseableArgument(NewClassTree syntaxNode, List<SymbolicValue> arguments) {
+    Iterator<SymbolicValue> iterator = arguments.iterator();
+    for (ExpressionTree argument : syntaxNode.arguments()) {
+      if (!iterator.hasNext()) {
+        throw new IllegalStateException("Mismatch between declared constructor arguments and argument values!");
+      }
+      final Type type = argument.symbolType();
+      final SymbolicValue value = iterator.next();
+      if (isCloseable(type)) {
+        return new CloseableArgument(needsClosing(type) ? value : null);
+      }
+    }
+    return null;
+  }
+
+  private static boolean isOpeningResource(NewClassTree syntaxNode) {
+    if (isWithinTryHeader(syntaxNode)) {
+      return false;
+    }
+    final Type type = syntaxNode.symbolType();
+    return needsClosing(type);
+  }
+
+  private static boolean needsClosing(Type type) {
+    for (String ignoredTypes : IGNORED_CLOSEABLE_SUBTYPES) {
+      if (type.is(ignoredTypes)) {
+        return false;
+      }
+    }
+    return isCloseable(type);
+  }
+
+  private static boolean isCloseable(Type type) {
+    return type.isSubtypeOf(JAVA_IO_CLOSEABLE);
+  }
+
+  private ProgramState closeResource(ProgramState programState, final SymbolicValue target) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Private method 'closeResource' is never used. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)